dash/include/dash/halo/Halo.h

```diff
@@ -294,8 +476,21 @@ class HaloSpec {
   constexpr HaloSpec(const Specs_t& specs) : _specs(specs) {}
 
   template <typename StencilSpecT>
-  HaloSpec(const StencilSpecT& stencil_specs) {
-    for(const auto& stencil : stencil_specs.specs()) {
+  HaloSpec(const StencilSpecT& stencil_spec) {
+    for(const auto& stencil : stencil_spec.specs()) {
+      auto stencil_combination = stencil;
+
+      set_region_spec(stencil_combination);
+      while(next_region(stencil, stencil_combination)) {
+        set_region_spec(stencil_combination);
+      }
+    }
```

Comment:
The code above seems to be duplicated below...

Reply:
solved